src/plugins/twitter/twitter.py

```diff
@@ -90,6 +91,34 @@ def _build(self, event_data, uri_id, origin, event_dict):
 
         return event_dict, events
 
+    def rate_limited_search(self, tso):
+        """ Search Twitter, sleeping when rate limitations are reached.
+
+        Args:
+            tso (TwitterSearchOrder): Search parameters for Twitter.
+
+        Returns:
+            Iterable: Tweets matching search parameters.
+        """
+        try:
+            return self.client.search_tweets_iterable(tso)
+
+        except TwitterSearchException as e:
+            if e.code != 429:
+                raise e
+
+        reset_datetime = datetime.fromtimestamp(
+            int(self.client.get_metadata()['x-rate-limit-reset'])
+        )
+        delta = reset_datetime - datetime.now()
+
+        logger.error(
+            f'Twitter rate limit reached. Sleeping {delta.seconds} seconds.'
+        )
+        time.sleep(delta.seconds + 10)
+
+        return self.client.search_tweets_iterable(tso)
```

Comment:
I am not fully convinced this is the best approach. In order to get the amount of seconds we need to wait, we add 10 seconds to the value in `x-rate-limit-reset`. What happens if Twitter sets `x-rate-limit-reset` to a point in time very far in the future?

Reply:
Yes, it's not ideal, but until the rate-limit window has been reset we will not be able to make requests to the Twitter API so I'm not sure there is a better way to do this.  

Currently, Twitter sets the request windows to 15 minutes long. So we should never be waiting longer than that.

Reply:
Correct me if I am wrong, but this implies waiting 15 minutes within the Celery task?

Reply:
No, you are correct. The task would wait 15 minutes for every 180 entries being processed. 

I'll look into a better way of doing this.

Reply:
http://docs.celeryproject.org/en/latest/userguide/tasks.html#Task.retry_backoff